fix: update upgrading_an_app page for JHipster 8

timothystone opened the pull request:
Use Font Awesome's Level Up Glyph over the "refresh" glyph, for semantics.

Use Markdown friendly Headers and Header links over HTML.

Fix the options available since JHipster 8.

Use Markdown syntax fences and link to updated Git Graph image (using Mermaid) and provide more stable path for editing with GitHub public Gist.

Standardize on the use of asterisks over mixed dash|asterisk unordered Markdown lists.

Add NetBeans ignore option (yes, we are still out here... ;))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/jhipster/jhipster.github.io?pullRequest=1326) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/jhipster/jhipster.github.io?pullRequest=1326) before we can accept your contribution.<br/><hr/>**Timothy Stone** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/jhipster/jhipster.github.io?pullRequest=1326) it.</sub>

atomfrede commented on nbproject/project.properties:
```diff
@@ -0,0 +1,3 @@
+files.encoding=UTF-8
```
I know it doesn't hurt, but should the nbproject folder really under version control?

timothystone-knsl replied:
Since I don't see other IDE meta-data, I can push a change that ignores the whole folder.

If I may make my case, my mind was changed when I was introduced to `gitignore.io`'s templates for IntelliJ, NetBeans, and many others. Now, I could see the logic that went into the exclusions provided by the templates. For example, `.idea/codeStyle.xml` was not excluded. My interest piqued and on inspection the logic made sense: you want to share the code style with your peers, not have your peers be instructed to set a style or have a commit or pipeline check "correct them."

Finally, note that`.nbproject/private` is excluded. IMHO there are aspects of many IDE meta-data folders that should be version controlled, but are generally excluded as a whole because determining which aspects should be excluded (ignored) and other aspects that should be included is necessarily difficult to dedicate time too. gitignore.io really changed my outlook.

timothystone replied:
@atomfrede removed the NB meta-data and updated the exclusions.

atomfrede commented on pages/upgrading_an_app.md:
```diff
@@ -35,30 +35,31 @@ For this sub-generator to work you need to have `git` installed from [http://git
 
 To upgrade your application, type:
 
 `npx jhipster upgrade`
```
Not sure if this is still valid from jh8 and up, as we have extracted the update/migrate into a dedicated blueprint. https://github.com/mshima/generator-jhipster-migrate

Maybe @mraible can tell more as I have never used the upgrade mechanism.

timothystone replied:
Is this just intended to be a simple prompt that you can't perform an upgrade from an unrelated project folder? I only want to make it more explicit for semantics, `myapplication` ... which one of _my applications_?

atomfrede reviewed (commented):
Thanks for the contribution, besides minor comments it looks good to me.

timothystone-knsl commented:
> Regarding my public gist for the Git graph. It's more stable than the JSFiddle that is currently provided, read: it doesn't work.
> 
> @atomfrede I think it's probably prudent to fork or replicate my Gist to the JHipster organization or member, or commit it as part of this repository. I didn't find an appropriate place for it off hand. `images`? It is an `image-as-code`. [GHFM is actually capable of displaying mermaid files](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/). But the Markdown parser of `jhipster.tech` may not appreciate it... though worth a test?

Testing a mermaid code fence...

mshima commented on images/upgrade_gitgraph.png:
- the jhipster_upgrade branch is not related to the main, you can start the line without splitting the main branch.
- each `npx jhipster upgrade` generates:
1. a commit at jhipster_upgrade with the current jhipster version
2. a merge commit at main (unrelated histories commit, a code equivalence) so following this merge, a commit applied to upgrade_branch will be applied as it is in main branch.
3. a commit at jhipster_upgrade with the updated jhipster version.
4. a merge commit at main that will apply commit 3 at main branch.

- you can split the `npx jhipster upgrade` commit in 2 commits, add a merge with a different color.

timothystone-knsl replied:
Thanks @mshima. I'll take a look at these changes. The "new graph" is mostly intended to correct the "old graph" going back to JHipster 3, and more importantly add the graph-as-code to the repository so as it does not exist in an individual's "fiddle" account (whether that individual is affiliated with JHipster or not).

timothystone replied:
@mshima i'm not quite sure what you are looking for (though I understand the words). The mermaid file is now included in the repository for experimentation. I only updated the existing graph for 8.x and I wanted to make sure that JHipster didn't rely on a "user account" with JSFiddle or GitHub for the content. The JSFiddle previously used is broken.

timothystone replied:
i'm working on these. I think I got the gist of the workflow and a handle on some of Mermaid's customizations.

timothystone replied:
@mshima is this more to your liking?

<img width="1647" alt="image" src="https://github.com/user-attachments/assets/f1d35d88-12e2-4bb5-85c9-a4748b81a493">

timothystone replied:
Updated image... 

![image](https://github.com/user-attachments/assets/04bf32b2-29c3-41a2-ae24-8af14dd6fa1d)

mraible commented:
I'd be happy to help move this along if you're still interested in helping @timothystone-knsl!

timothystone-knsl commented:
> I'd be happy to help move this along if you're still interested in helping @timothystone-knsl!

I'm still interested @mraible

mraible commented:
@timothystone Can you please fix conflicts if this page still needs to be updated? I tried using the [current page](https://www.jhipster.tech/upgrading-an-application/) to update my jhipster8-demo app last night and it worked flawlessly!

timothystone commented:
> @timothystone Can you please fix conflicts if this page still needs to be updated? I tried using the [current page](https://www.jhipster.tech/upgrading-an-application/) to update my jhipster8-demo app last night and it worked flawlessly!

 Resolved conflicts and pushed, then realized I should not have rebased (as a rule)... hope that hasn't cause anyone any headaches.

mraible commented on pages/upgrading_an_app.md:
```diff
@@ -102,35 +102,36 @@ On the first run of the JHipster upgrade sub-generator, in order to avoid erasin
-- There are some helpful approaches from the JHipster community around designing the application in such a way that makes the update process easier, and reduces the amount of merge conflicts. We recommend using [JHipster Side-by-Side approach](https://www.youtube.com/watch?v=Gg5CYoBdpVo).  
+* There are some helpful approaches from the JHipster community around designing the application in such a way that makes the update process easier, and reduces the amount of merge conflicts. We recommend using [JHipster Side-by-Side approach](https://www.youtube.com/watch?v=Gg5CYoBdpVo).  
 
-<h2 id="manual_upgrade">Option 2 - manual upgrade</h2>
```
I believe the id existed so there would be an anchor to this section of the page, which might be used in another file.

mraible commented on pages/upgrading_an_app.md:
```diff
@@ -70,7 +70,7 @@ Here is how the upgrade process works graphically (read the sections below to ha
 ![GitGraph]({{ site.url }}/images/upgrade_gitgraph.png)
 
-(this image comes from [JSFiddle](http://jsfiddle.net/lordlothar99/tqp9gyu3/) )
+(this image comes from Mermaid [GitHub Gist](https://gist.github.com/timothystone/aa5bcd27d1ede5a08a7065e2e1a1cb77) )
```
Can you get a high-res version of this image and update the `upgrade_gitgraph.png` file? It looks a little grainy in the preview: https://deploy-preview-1326--jhipster-site.netlify.app/upgrading-an-application/

mraible commented:
We updated the website with a new design. Can you please update this issue accordingly?

timothystone commented:
> We updated the website with a new design. Can you please update this issue accordingly?

Done. Still owe you a high resolution image.

timothystone commented:
> > We updated the website with a new design. Can you please update this issue accordingly?
> 
> Done. Still owe you a high resolution image.

Done and Done. added the raw mermaid file to the static resources to move away from user accounts on GitHub and JSFiddle. The code is in the repository.

mraible commented:
Is it possible to make the graphical view image clickable so a reader can zoom in? I'm on an iPad and the font is a little small.

timothystone commented:
> Is it possible to make the graphical view image clickable so a reader can zoom in? I'm on an iPad and the font is a little small.

Depends on what kind of lightbox support docusaurus has built-in. I'll poke around.

timothystone-knsl commented:
@mraible asking @mshima to review the image. [Marcelo asked](https://github.com/jhipster/jhipster.github.io/pull/1326#discussion_r1496613668) upthread for some nuance and I'm trying to bend Mermaid to my will. 🧜‍♀️

timothystone commented:
@mraible Without added functionality, i.e., a lightbox, I can't get the image to link to open in a new window or page. 🙁

mraible commented:
Can't you just link to the image file? That should only require good ol' HTML.

timothystone commented:
> Can't you just link to the image file? That should only require good ol' HTML.

Tried that. The image didn't render... something to do with docusaurus? I'm going to push a version with the lightbox, just in case you want to consider it.

timothystone commented:
I must say I like that... looking on my iPad... looks good.

timothystone commented:
@mraible the plugin does effect all images in the site. I did configure Docusaurus [to allow for image exclusion though](https://www.npmjs.com/package/plugin-image-zoom#excluding-images-from-using-zoom).

mraible commented:
@timothystone Looks good! I'm on vacation for the long weekend and only looked at the preview on my phone. If someone can give a more thorough review/merge, please do!

DanielFran reviewed (approved):
Look nice in mobile too

DanielFran commented:
Thanks @timothystone, great work
